calico_node/startup/startup.go

```diff
@@ -781,126 +873,147 @@ func ensureDefaultConfig(cfg *api.CalicoAPIConfig, c *client.Client, node *api.N
 	// for containers to speak to services running on the host (e.g. k8s
 	// pods speaking to k8s api-server, and mesos tasks registering with agent
 	// on startup).  Note: KDD does not yet support per-node felix config.
-	if cfg.Spec.DatastoreType != api.Kubernetes {
-		if err := ensureNodeFelixConfig(c, node.Metadata.Name, "DefaultEndpointToHostAction", "RETURN"); err != nil {
-			return err
+	if cfg.Spec.DatastoreType != apiconfig.Kubernetes {
```

Comment:
So was the outcome of the discussions that we always need a per-node config too?  I was rather hoping we wouldn't need that.

Reply:
Seems like one reason, as Casey pointed out, was for hybrid deployments where you'd want to have a different per node config for k8s nodes (different from OpenStack)